target/linux/ipq40xx/patches-4.14/090-ipq40xx-fix-high-resolution-timer.patch

```diff
@@ -0,0 +1,30 @@
+From 09f145f417a5d64d6b8d4476699dfb0eccc6c784 Mon Sep 17 00:00:00 2001
+From: dissent1 <be.dissent@gmail.com>
+Date: Tue, 7 May 2019 10:14:05 +0300
+Subject: [PATCH] ipq40xx: fix high resolution timer
+
+Cherry-picked from CAF QSDK repo.
+Original commit message:
+The kernel is failing in switching the timer for high resolution
+mode and clock source operates in 10ms resolution. The always-on
+property needs to be given for timer device tree node to make
+clock source working in 1ns resolution.
+
+Change-Id: I7c00b3c74d97c2a30ac9f05e18b511a0550fd459
+Signed-off-by: Abhishek Sahu <absahu@codeaurora.org>
+
+Signed-off-by: Pavel Kubelun <be.dissent@gmail.com>
+---
+ arch/arm/boot/dts/qcom-ipq4019.dtsi | 1 +
+ 1 file changed, 1 insertion(+)
+
+--- a/arch/arm/boot/dts/qcom-ipq4019.dtsi
++++ b/arch/arm/boot/dts/qcom-ipq4019.dtsi
+@@ -167,6 +167,7 @@
+ 			     <1 4 0xf08>,
+ 			     <1 1 0xf08>;
+ 		clock-frequency = <48000000>;
++		always-on;
```

Comment:
without (sysrq-q):

```
[   37.079417] Timer List Version: v0.8
[   37.086807] HRTIMER_MAX_CLOCK_BASES: 4
[   37.089470] now at 37076096318 nsecs
[   37.092997]
[   37.096706] cpu: 0
[   37.098151]  clock 0:
[   37.099978]   .base:       cfdb4440
[   37.102319]   .index:      0
[   37.105645]   .resolution: 10000000 nsecs
[   37.108742]   .get_time:
[   37.108753] ktime_get
```

vs with:

```
[12912.405691] Timer List Version: v0.8
[12912.412521] HRTIMER_MAX_CLOCK_BASES: 4
[12912.415763] now at 12912484526672 nsecs
[12912.420041] 
[12912.422989] cpu: 0
[12912.424688]  clock 0:
[12912.426514]   .base:       c7cb4440
[12912.428857]   .index:      0
[12912.432190]   .resolution: 1 nsecs
[12912.435280]   .get_time:   
[12912.435289] ktime_get
[12912.438490] 
```

:+1: Have you sent this upstream?

Reply:
Nope, but I guess it should be upstreamed along with opp voltage and sleep clock when the confusion is resolved :)